gateway/slack.go

```diff
@@ -407,35 +407,47 @@ func (sc *SlackClient) Poop(chans *ClientChans) {
 			case "message":
 				messageData := event.Data.(*slack.MessageEvent)
 
-				switch messageData.SubType {
-				case "":
-					user, err := sc.ResolveUser(messageData.User)
-					if err != nil {
-						log.Println(err)
-						continue
+				var sender *SlackUser
+				if messageData.User != "" {
+					var err error
+					if sender, err = sc.ResolveUser(messageData.User); err != nil {
+						log.Printf("could not resolve user for message [%v]: %+v", err, messageData)
 					}
+				}
 
-					var messageEventTarget string
+				var target string
+				if messageData.Channel != "" {
 					if isDmChannel(messageData.Channel) {
-						target, err := sc.ResolveDMToUser(messageData.Channel)
-						if err != nil {
-							fmt.Printf("error resolving DM: %v\n", err)
-							continue
+						if targetUser, err := sc.ResolveDMToUser(messageData.Channel); err != nil {
+							log.Printf("could not resolve DM target for message [%v]: %+v", err, messageData)
+						} else {
+							target = targetUser.Nick
 						}
-
-						messageEventTarget = target.Nick
-
 					} else {
-						channel, err := sc.ResolveChannel(messageData.Channel)
-						if err != nil {
-							log.Println(err)
-							continue
+						if channel, err := sc.ResolveChannel(messageData.Channel); err != nil {
+							log.Printf("could not resolve channel for message [%v]: %+v", err, messageData)
+						} else {
+							target = channel.Name
```

Comment:
Likewise, if target resolution fails, should we fail or have a fallback? Or are we going to let the IRC client try to turn this into a real message and fail there?

Reply:
hmm, should probably actually break out here yeah